apps/hostmetrics.go

```diff
@@ -367,6 +386,19 @@ func (r MetricsReceiverHostmetrics) Pipelines(ctx context.Context) []otel.Receiv
 	return pipelines
 }
 
+// TODO:
+func canTurnOffReceiver(expectedMetrics []string, metricsPattern []string) bool {
+	matched := set.Set[string]{}
+	for _, m := range expectedMetrics {
+		for _, rp := range metricsPattern {
+			if r, _ := regexp.MatchString(strings.ReplaceAll(rp, "*", ".*"), m); r {
```

Comment:
Why not make this a method on the `MetricsProcessorExcludeMetrics`?

Don't use regexp for this, because that will cause all sorts of weird behavior with metacharacters like `.`. Just use `filepath.Match(rp, m)`, and it can be as simple as:

```
func (p MetricsProcessorExcludeMetrics) AllMetricsExcluded(metrics ...string) bool {
	OUTER:
	for _, metric := range metrics {
		for _, excludePattern := range p.MetricsPattern {
			if filepath.Match(excludePattern, metric) {
				continue OUTER
			}
		}
		return false
	}
	return true
}
```

Reply:
Hm, yeah, good catch on the different behavior of `*`. Refactoring to use the same logic makes sense.